lib/src/stream_sink_transformer/reject_errors.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// A [StreammSink] wrapper that rejects all errors passed into the sink.
+class RejectErrorsSink<T> implements StreamSink<T> {
+  /// The inner sink being wrapped.
+  final StreamSink<T> _inner;
+
+  @override
+  Future<void> get done => _doneCompleter.future;
```

Comment:
I'd move the getters below the constructor.
(State above constructor, members below, makes it easier to figure out the state of the object).

Reply:
This is just the public interface for `_doneCompleter`, which is state. I can move the `canceled` getter below the constructor, although I find it useful to group getters with fields because they both *logically* (if not *physically*) represent state.

Reply:
Let's say "internal state" then. When trying to understand a class, as a programmer working on it, I like to have the internal state in one place, and by itself.

I'm aware others do it differently, and the style guide is silent on member ordering.